Support/Image.swift

```diff
@@ -0,0 +1,91 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import TensorFlow
+
+public struct Image {
+    public enum ByteOrdering {
+        case bgr
+        case rgb
+    }
+
+    enum ImageTensorFormat {
+        case float(data: Tensor<Float>)
+        case uint8(data: Tensor<UInt8>)
+    }
+
+    let imageData: ImageTensorFormat
+
+    public init(tensor: Tensor<UInt8>) {
+        self.imageData = .uint8(data: tensor)
+    }
+
+    public init(tensor: Tensor<Float>) {
+        self.imageData = .float(data: tensor)
+    }
+
+    public init(jpeg url: URL, byteOrdering: ByteOrdering = .rgb) {
+        let loadedFile = Raw.readFile(filename: StringTensor(url.absoluteString))
+        let loadedJpeg = Raw.decodeJpeg(contents: loadedFile, channels: 3, dctMethod: "")
+        if byteOrdering == .bgr {
+            self.imageData = .uint8(
+                data: Raw.reverse(loadedJpeg, dims: Tensor<Bool>([false, false, false, true])))
+        } else {
+            self.imageData = .uint8(data: loadedJpeg)
+        }
+    }
+
+    public func save(to url: URL, quality: Int64 = 95) {
+        // Currently only saving in grayscale 
+        let outputImageData: Tensor<UInt8>
+        switch self.imageData {
+        case let .uint8(data): outputImageData = data
+        case let .float(data):
+            let lowerBound = data.min(alongAxes: [0, 1])
+            let upperBound = data.max(alongAxes: [0, 1])
+            let adjustedData = (data - lowerBound) * (255.0 / (upperBound - lowerBound))
+            outputImageData = Tensor<UInt8>(adjustedData)
+        }
+
+        let encodedJpeg = Raw.encodeJpeg(
+            image: outputImageData, format: .grayscale, quality: quality, xmpMetadata: "")
+        Raw.writeFile(filename: StringTensor(url.absoluteString), contents: encodedJpeg)
+    }
+
+    public func resized(to size: (Int, Int)) -> Image {
+        switch self.imageData {
+        case let .uint8(data):
+            return Image(
+                tensor: Raw.resizeBilinear(
+                    images: Tensor<UInt8>([data]),
+                    size: Tensor<Int32>([Int32(size.0), Int32(size.1)])))
+        case let .float(data):
+            return Image(
+                tensor: Raw.resizeBilinear(
+                    images: Tensor<Float>([data]),
+                    size: Tensor<Int32>([Int32(size.0), Int32(size.1)])))
+        }
+
+    }
+}
+
+public func saveImage(tensor: Tensor<Float>, size: (Int, Int), directory: String, name: String) {
```

Comment:
```suggestion
public func saveImage(_ tensor: Tensor<Float>, size: (Int, Int), directory: String, name: String) {
```
The `tensor` argument label is merely repeating the argument type information, so it should be omitted according to the Swift API Design Guidelines.